great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -441,13 +441,15 @@ def run(
         self,
         variables: Optional[Dict[str, Any]] = None,
         rules: Optional[Dict[str, Dict[str, Any]]] = None,
+        **kwargs: dict,
```

Comment:
@cdkini It has to be documented, just like in the `Scope` of this PR.

Reply:
we need to get away from all these dictionaries, especially since we don't expect anyone to call these methods but us. No more kwargs!

We don't have to call these "RuntimeEnvironment" any more, they are no longer simple settings, they are "ColumnDomainDirectives". If we want to generalize it, we could say that ColumnDomainDirectives are just one type of Directive that could be passed to DataAssistants (and the Profiler). This makes it obvious we should build a lot more smarts/functionality into the type, and I think makes it easier for the user to understand what they are being asked to supply.

Significantly, the primary user of this IS US, the developers of great_expectations. Convenience methods for those that like dictionaries can be provided at a higher level, and reconciled as Directives by time we get to the actual DataAssistant.

And for those who come along to implement their own DataAssistants later, they will really appreciate knowing that this parameter is a way of providing Directives, vs. a bag of nondescript values

Reply:
Are `include_column_names` and `exclude_column_names` available for every Data Assistant? If so, does it make sense to add these here separately from `kwargs` to make that clear for the users when they use auto-complete? If not, does it make sense to add these to the Data Assistants that use them?